packages/backend/src/nest/registration/registration.service.ts

```diff
@@ -72,10 +67,50 @@ export class RegistrationService extends EventEmitter implements OnModuleInit {
     }
   }
 
-  public async registerUserCertificate(csr: string): Promise<void> {
-    const result = await issueCertificate(csr, this._permsData)
-    if (result?.cert) {
-      this.emit(RegistrationEvents.NEW_USER, { certificate: result.cert })
+  // Apparently, JS will run each function to completion. So assuming
+  // we do not have multiple threads, this function should run to
+  // completion before it is called again. Because we take the CSRs
+  // and then get the latest view of certificates, filter CSRs and
+  // then update certificates in a single function, we should never
+  // issue a CSR that is contained in the certificates list, at least
+  // in this function... as long as there is no other code that
+  // updates the certificates list. Something else to consider is that
+  // because this function is called asynchronously, there may be
+  // several invocations with different CSRs and they may run in an
+  // unexpected order. We could address that if it's an issue, but I
+  // think that might only affect the order of CSR registration.
+  public async registerUserCertificates(payload: { csrs: string[]; id?: string }) {
+    if (!this.storageService) {
+      throw new Error("Storage Service must be initialized before the Registration Service")
     }
+
+    // Lack of permsData means that we are not the owner of the
+    // community in the official model of the app, however anyone can
+    // modify the source code, put malicious permsData here, issue
+    // false certificates and try to trick other users. To prevent
+    // that, peers verify that anything that is written to the
+    // certificate store is signed by the owner.
+    if (!this._permsData) {
```

Comment:
Why is this here? Why would we trigger a "finished issuing certificates" event if we are not the owner?

Reply:
This statement basically says "You're not the owner? Do not continue issues certificates

Reply:
The name is misleading. AFAIK it's only used for resolving promise in csrReplicatedPromiseMap in storage.